RInAppMessaging/Classes/InAppMessagingModule.swift

```diff
@@ -121,10 +121,10 @@ internal class InAppMessagingModule: AnalyticsBroadcaster,
 
     private func flushEventBuffer(discardEvents: Bool) {
         if !discardEvents {
-            eventBuffer.forEach {
-                eventMatcher.matchAndStore(event: $0)
+            eventBuffer.forEach { event in
+                self.eventMatcher.matchAndStore(event: event)
+                self.campaignTriggerAgent.validateAndTriggerCampaigns()
```

Comment:
What's the benefit to bringing this into the loop? Does it fix a problem?

Reply:
Yes. because validator needs to go though the campaign list for each event. those 2 lines are just like a single `logEvent`. I could change the logic inside validator instead. Not sure what's the best

Reply:
No it's fine.